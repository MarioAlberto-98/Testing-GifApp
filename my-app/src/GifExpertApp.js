import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

//{defaultCategories = []}


//Codigo Original
/*
export const GifExpertApp = () => {
    
    const [categories, setCategories] = useState(['One Punch']);

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories } />
            <hr />

            <ol>
                {
                    categories.map( category  => (
                        <GifGrid 
                            key={ category }
                            category={ category }
                        />
                    ))
                }
            </ol>

        </>
    )
}*/


//Codigo para testing

export const GifExpertApp = ({defaultCategories = []}
    ) => {
    
    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories } />
            <hr />

            <ol>
                {
                    categories.map( category  => (
                        <GifGrid 
                            key={ category }
                            category={ category }
                        />
                    ))
                }
            </ol>

        </>
    )
}

