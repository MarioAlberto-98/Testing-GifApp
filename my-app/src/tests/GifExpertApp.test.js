import React from 'react';
import { shallow } from 'enzyme';
import {GifExpertApp} from '../GifExpertApp';
describe('Prueba en <GifExpertApp />', ()=>{

    test('debe mostrarse correctamente', () =>{
        const wrapper =shallow(<GifExpertApp />)
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrart una lista de categorias', () => {
        const  categories=['One Punch', 'Dragon Ball' ];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })


})