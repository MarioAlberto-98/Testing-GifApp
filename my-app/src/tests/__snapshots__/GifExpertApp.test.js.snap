// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifExpertApp /> debe de mostrart una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Prueba en <GifExpertApp /> debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
